### PR TITLE
Add markers for remote tests

### PR DESCRIPTION
**Pull request by dmitrypolo**
This addresses #543.

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+markers =
+    ci_vendor: Only run these tests within the CI Vendor environment.
```
Comment (whimboo) [inline pytest.ini]: Mind renaming this marker to `ci_only`? It better describes what it is used for. Also please fix-up the sentence and remove the `vendor`.

File: tests/conftest.py [hunk 2]
```diff
@@ -16,3 +16,12 @@ def httpd():
     httpd.start(block=False)
     yield httpd
     httpd.stop()
+
+
+def pytest_runtest_setup(item):
+    valid_ci = os.getenv('CI', False)
```
Comment (whimboo) [inline tests/conftest.py]: Mind renaming to `ci_enabled`?

File: pytest.ini (hunk 1 above)
Comment (dmitrypolo) [reply pytest.ini]: done

File: tests/conftest.py (hunk 2 above)
Comment (dmitrypolo) [reply tests/conftest.py]: done

**Comment by dmitrypolo** [comment]
@whimboo changed per your comments

**Comment by whimboo** [comment]
I fixed two small things, and pushed those commits to your branch. It looks all fine now, so I will merge it. Thanks a lot for getting it added!

**Pull request merged**